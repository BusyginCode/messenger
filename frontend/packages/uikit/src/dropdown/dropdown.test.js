import React from 'react'
import { shallow } from 'enzyme'
import Dropdown from './dropdown'

describe('uikit/Dropdown', () => {
  it('should render correctly', () => {
    expect(
      shallow(
        <Dropdown overlay={<div>overlay</div>}>
          <div>trigger</div>
        </Dropdown>
      )
    ).toMatchSnapshot()
  })

  it('should render with options correctly', () => {
    expect(
      shallow(
        <Dropdown options={[{ id: 1 }]}>
          <div>trigger</div>
        </Dropdown>
      )
    ).toMatchSnapshot()
  })
})
