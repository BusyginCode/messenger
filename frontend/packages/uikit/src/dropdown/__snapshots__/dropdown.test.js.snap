// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uikit/Dropdown should render correctly 1`] = `
<Dropdown
  defaultVisible={false}
  hideAction={Array []}
  minOverlayWidthMatchTrigger={true}
  onVisibleChange={[Function]}
  overlay={
    <div>
      overlay
    </div>
  }
  overlayClassName=""
  overlayStyle={Object {}}
  placement="bottomLeft"
  prefixCls="rc-dropdown"
  showAction={Array []}
  trigger={
    Array [
      "click",
    ]
  }
>
  <div>
    trigger
  </div>
</Dropdown>
`;

exports[`uikit/Dropdown should render with options correctly 1`] = `
<Dropdown
  defaultVisible={false}
  hideAction={Array []}
  minOverlayWidthMatchTrigger={true}
  onVisibleChange={[Function]}
  overlay={
    <Overlay
      options={
        Array [
          Object {
            "id": 1,
          },
        ]
      }
    />
  }
  overlayClassName=""
  overlayStyle={Object {}}
  placement="bottomLeft"
  prefixCls="rc-dropdown"
  showAction={Array []}
  trigger={
    Array [
      "click",
    ]
  }
>
  <div>
    trigger
  </div>
</Dropdown>
`;
